src/home/celebration/NftReward.tsx

```diff
@@ -0,0 +1,245 @@
+import i18n from 'src/i18n'
+import { nftsWithMetadataSelector } from 'src/nfts/selectors'
+import Colors from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import { formatDistanceToNow } from 'src/utils/time'
+
+export default function NftRewardBottomSheet() {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  const canShowNftReward = useSelector(showNftRewardSelector)
+  const celebratedNft = useSelector(celebratedNftSelector)
+
+  const nfts = useSelector(nftsWithMetadataSelector)
+  const matchingNft = useMemo(
+    () =>
+      nfts.find(
+        (nft) =>
+          !!celebratedNft &&
+          !!celebratedNft.networkId &&
+          celebratedNft.networkId === nft.networkId &&
+          !!celebratedNft.contractAddress &&
+          celebratedNft.contractAddress === nft.contractAddress
+      ),
+    [celebratedNft]
+  )
+
+  const insets = useSafeAreaInsets()
+  const insetsStyle = { paddingBottom: Math.max(insets.bottom, Spacing.Regular16) }
+
+  const bottomSheetRef = useRef<BottomSheetRefType>(null)
+
+  const nftCelebration = useSelector(nftCelebrationSelector)
+  const expirationDate = new Date(nftCelebration?.expirationDate ?? 0)
+  const reminderDate = new Date(nftCelebration?.reminderDate ?? 0)
+  const deepLink = nftCelebration?.deepLink ?? ''
+
+  const { expirationStatus, expirationLabelText } = useExpirationStatus(
+    expirationDate,
+    reminderDate
+  )
+
+  const isVisible = canShowNftReward && matchingNft
+
+  useEffect(() => {
+    if (isVisible) {
+      // Wait for the home screen to have less ongoing async tasks.
+      // This should help the bottom sheet animation run smoothly.
+      const timeoutId = setTimeout(() => {
+        bottomSheetRef.current?.expand()
+      }, 1000)
+      return () => clearTimeout(timeoutId)
+    }
+  }, [isVisible])
+
+  const handleBottomSheetPositionChange = (index: number) => {
+    if (!celebratedNft) {
+      return // This should never happen
+    }
+
+    if (index === -1) {
+      ValoraAnalytics.track(HomeEvents.nft_reward_dismiss, {
+        networkId: celebratedNft.networkId,
+        contractAddress: celebratedNft.contractAddress,
+        remainingDays: differenceInDays(expirationDate, Date.now()),
+      })
+
+      dispatch(nftRewardDisplayed())
+    }
+  }
+
+  const handleCtaPress = () => {
+    if (!celebratedNft) {
+      return // This should never happen
+    }
+
+    ValoraAnalytics.track(HomeEvents.nft_reward_accept, {
+      networkId: celebratedNft.networkId,
+      contractAddress: celebratedNft.contractAddress,
+      remainingDays: differenceInDays(expirationDate, Date.now()),
+    })
+
+    bottomSheetRef.current?.close()
+
+    const isSecureOrigin = true
+    dispatch(openDeepLink(deepLink, isSecureOrigin))
+  }
+
+  const pillLabel =
+    expirationStatus === ExpirationStatus.expired
+      ? t('nftCelebration.rewardBottomSheet.expired')
```

Comment:
if the reward is expired, should we hide the sheet rather than show a user that they've got expired rewards?

Reply:
yes, hiding the sheet could be an option. with it, we have one string less to translate.